libtsuba/include/tsuba/TxnContext.h

```diff
@@ -0,0 +1,92 @@
+#ifndef KATANA_LIBTSUBA_TSUBA_TXNCONTEXT_H_
+#define KATANA_LIBTSUBA_TSUBA_TXNCONTEXT_H_
+
+namespace tsuba {
+
+class KATANA_EXPORT TxnContext {
+public:
+  TxnContext(){};
+  ~TxnContext() = default;
+
+  void InsertNodePropertyRead(std::string name) {
+    node_properties_read.push_back(name);
+  }
+
+  void InsertNodePropertyRead(std::vector<std::string> names) {
+    node_properties_read.insert(
+        node_properties_read.end(), names.begin(), names.end());
+  }
+
+  void InsertNodePropertyWrite(std::string name) {
+    node_properties_write.push_back(name);
+  }
+
+  void InsertNodePropertyWrite(std::vector<std::string> names) {
+    node_properties_write.insert(
+        node_properties_write.end(), names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyRead(std::string name) {
+    edge_properties_read.push_back(name);
+  }
+
+  void InsertEdgePropertyRead(std::vector<std::string> names) {
+    edge_properties_read.insert(
+        edge_properties_read.end(), names.begin(), names.end());
+  }
+
+  void InsertEdgePropertyWrite(std::string name) {
+    edge_properties_write.push_back(name);
+  }
+
+  void InsertEdgePropertyWrite(std::vector<std::string> names) {
+    edge_properties_write.insert(
+        edge_properties_write.end(), names.begin(), names.end());
+  }
+
+  void SetAllPropertiesRead() { all_properties_read = true; }
+
+  void SetAllPropertiesWrite() { all_properties_write = true; }
```

Comment:
Again in the flatbuffer `RDGAccess`, there are types of access like `AllProperties` and `Topology`, meaning that some operation could possibly change all the properties (logically, not by inserting one by one. Right now I'm not sure where it can happen but it's defined in `RDGAccess` so correspondent things needs to be added here) and topology (I saw some topology changing code, but haven't investigated in details). With those codes, `SetAllProperties...()` and `SetAllTopology()` needs to be called, same as `SetNodeProperty...()`.